plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -2949,6 +2949,23 @@ private KVMPhysicalDisk getPhysicalDiskPrimaryStore(PrimaryDataStoreTO primaryDa
         return storagePool.getPhysicalDisk(data.getPath());
     }
 
+    /**
+     * Check if IO_URING is supported by qemu
+     */
+    protected boolean isIoUringSupportedByQemu() {
+        s_logger.debug("Checking if iouring is supported");
+        String hostOsKey = "Host.OS";
+        Map<String, String> versionStrings = getVersionStrings();
+        if (MapUtils.isEmpty(versionStrings) || !versionStrings.containsKey(hostOsKey)) {
+            return false;
+        }
+        String os = versionStrings.get(hostOsKey);
+        String qemuBinary = os.equalsIgnoreCase("ubuntu") ? "/usr/bin/qemu-system-x86_64" : "/usr/libexec/qemu-kvm";
```

Comment:
@nvazquez 
I think we can get the qemu emulator in other ways like
(1) use `virsh capabilities` to get the emulator for x86_64
(2) check if the files exist. e.g. if `/usr/bin/qemu-system-x86_64` exist, then use it, otherwise use `/usr/libexec/qemu-kvm`

(1) looks better.

Reply:
@weizhouapache I have been testing on ubuntu20 and it seems like the check does not pass, so I have refactored it to be enabled by default on ubuntu: cc @wido @GabrielBrascher @slavkap 
````
# /usr/share/cloudstack-common/scripts/vm/hypervisor/versions.sh
Host.OS=Ubuntu
Host.OS.Version=20.04
Host.OS.Kernel.Version=5.9.0-050900-generic
# ldd /usr/bin/qemu-system-x86_64 | grep -i uring
# /usr/bin/qemu-system-x86_64 -drive aio=uring
Unable to init server: Could not connect: Connection refused
qemu-system-x86_64: -drive aio=uring: invalid aio option

Reply:
This was regarding the qemu and libvirtd versions (< 5.0 and <6.3.0) - not an issue

Reply:
Refactored @weizhouapache